### PR TITLE
feat: update committee from the committee definions TXs

### PR DESCRIPTION
**Pull request by Cifko**
Description
---
Update the validation node based on the committee definitions.
Fixes the bug that I was not able to create asset.

How Has This Been Tested?
---
Manually (but only adding, since we don't have committee change option). But should work fine.

### CODE REVIEW

File: applications/tari_validator_node/src/dan_node.rs [hunk 1]
```diff
@@ -115,54 +122,64 @@ impl DanNode {
...
+                    monitoring.add_state(asset.public_key, mined_height, true);
+                }
+                let mut known_active_public_keys = assets.into_iter().map(|(asset, _)| asset.public_key);
+                let active_public_keys = monitoring
+                    .get_active_public_keys()
+                    .into_iter()
+                    .cloned()
+                    .collect::<Vec<PublicKey>>();
+                for public_key in active_public_keys {
+                    if !known_active_public_keys.contains(&public_key) {
```
Comment (sdbondi) [inline applications/tari_validator_node/src/dan_node.rs]: Rather use this and remove the `itertools` dependency
```suggestion
                    if !known_active_public_keys.any(|pk| pk == public_key) {
```

File: dan_layer/core/src/workers/consensus_worker.rs [hunk 2]
```diff
@@ -128,7 +134,7 @@ impl<TSpecification: ServiceSpecification<Addr = PublicKey>> ConsensusWorker<TSp
             "Consensus worker started for asset '{}'. Tip: {}", self.asset_definition.public_key, self.current_view_id
         );
         let starting_view = self.current_view_id;
-        loop {
+        while !stop.as_ref().load(Ordering::Relaxed) {
```
Comment (sdbondi) [inline dan_layer/core/src/workers/consensus_worker.rs]: nit: `.` operator dereferences the `Arc`

```suggestion
        while !stop.load(Ordering::Relaxed) {
```

File: applications/tari_validator_node/Cargo.toml [hunk 3]
```diff
@@ -32,6 +32,7 @@ blake2 = "0.9.2"
 clap = "2.33.3"
 digest = "0.9.0"
 futures = { version = "^0.3.1" }
+itertools = "0.10.3"
```
Comment (sdbondi) [inline applications/tari_validator_node/Cargo.toml]: Can remove this dependency (see other comment)

File: applications/tari_validator_node/src/asset.rs [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+    pub fn update_height<Fstart>(&mut self, height: u64, start: Fstart)
+    where Fstart: Fn(AssetDefinition) -> Arc<AtomicBool> {
+        if let Some((&height, &involment)) = self
+            .next_states
+            .iter()
+            .find(|(&mined_height, _)| mined_height <= height)
+        {
+            // State change
+            if self.current_state != involment {
```
Comment (sdbondi) [inline applications/tari_validator_node/src/asset.rs]:
```suggestion
            if self.current_state != involvement {
```

File: applications/tari_validator_node/src/asset.rs [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+};
+
+use tari_dan_core::models::AssetDefinition;
+
+#[derive(Debug)]
+pub struct Asset {
+    definition: AssetDefinition,
+    current_state: bool,
+    // Changes in the committee for this asset.
+    // Mined height of the change TXs, and the involvment in the committe (true = part of committee)
```
Comment (sdbondi) [inline applications/tari_validator_node/src/asset.rs]:
```suggestion
    // Mined height of the change TXs, and the involvement in the committee (true = part of committee)
```

File: applications/tari_validator_node/src/asset.rs [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+    // If we are part of committe or there is a next state then monitor this asset
+    pub fn should_monitor(&self) -> bool {
+        self.current_state || !self.next_states.is_empty()
+    }
+
+    pub fn add_state(&mut self, height: u64, involment: bool) {
+        self.next_states.insert(height, involment);
+    }
+
+    pub fn get_current_state(&self) -> bool {
```
Comment (sdbondi) [inline applications/tari_validator_node/src/asset.rs]: Maybe should be called `is_committee_member` or `is_managing_asset`

**Review by sdbondi — COMMENTED** [review]
LGTM - some comments

File: applications/tari_validator_node/src/asset.rs (hunk 6 above)
Comment (Cifko) [reply applications/tari_validator_node/src/asset.rs]: I like the `is_committee_member`.

File: dan_layer/core/src/workers/consensus_worker.rs (hunk 2 above)
Comment (Cifko) [reply dan_layer/core/src/workers/consensus_worker.rs]: Cool, didn't know that. There is still much to learn in rust for me :)

**Review by stringhandler — APPROVED** [review]

**Pull request merged**